projects/plugins/jetpack/extensions/blocks/ai-paragraph/edit.js

```diff
@@ -74,20 +77,60 @@ function getSuggestionFromOpenAI(
 		} );
 }
 
+function createPrompt( title = '', content = '', categoryNames = '' ) {
+	// If title is not added, we will only complete.
+	if ( ! title ) {
+		return content;
+	}
+
+	// Title, content and categories
+	if ( content && categoryNames.length ) {
+		return sprintf(
+			/** translators: This will be a prompt to OpenAI to generate a post based on the post title, comma-seperated category names and the last 240 characters of content. */
+			__( "This is a post titled '%1$s', published in categories '%2$s':\n\n … %3$s", 'jetpack' ), // eslint-disable-line @wordpress/i18n-no-collapsible-whitespace
+			title,
+			categoryNames,
+			content
+		);
+	}
+
+	// No content, only title and categories
+	if ( categoryNames.length ) {
+		return sprintf(
+			/** translators: This will be a prompt to OpenAI to generate a post based on the post title, and comma-seperated category names. */
+			__( "This is a post titled '%1$s', published in categories '%2$s':\n", 'jetpack' ), // eslint-disable-line @wordpress/i18n-no-collapsible-whitespace
+			title,
+			categoryNames
+		);
+	}
+
+	// Title and content
+	if ( content ) {
+		return sprintf(
+			/** translators: This will be a prompt to OpenAI to generate a post based on the post title, and the last 240 characters of content. */
+			__( "This is a post titled '%1$s':\n\n…%2$s", 'jetpack' ), // eslint-disable-line @wordpress/i18n-no-collapsible-whitespace
+			title,
+			content
+		);
+	}
+
+	return sprintf(
+		/** translators: This will be a prompt to OpenAI to generate a post based on the post title */
+		__( 'Write content of a post titled "%s"', 'jetpack' ),
+		title
+	);
+}
+
 /**
  * Gather data available in Gutenberg and prepare the best prompt we can come up with.
  *
  * @param {Function} select - function returning Gutenberg data store.
  * @returns {string} - prompt ready to pipe to OpenAI.
  */
 function preparePromptBasedOnEditorState( select ) {
-	const prompt = formatEditorCanvasContentForCompletion( select( 'core/block-editor' ) );
-
-	// If a user started typing something, we will just create a completion.
-	if ( prompt.length > 0 ) {
-		// We only take the last 240 chars into account, otherwise the prompt gets too long and because we have a 110 tokens limit, there is no place for response.
-		return prompt.slice( -240 );
-	}
+	const editorContent = formatEditorCanvasContentForCompletion(
+		select( 'core/block-editor' )
+	).slice( -240 );
```

Comment:
Would be nice to have that as an constant...

Reply:
There is plenty to improve here, I am just trying to ship as many big improvements before the next JP release cycle